nav2_amcl/src/sensors/laser/likelihood_field_model.cpp

```diff
@@ -35,7 +35,14 @@ LikelihoodFieldModel::LikelihoodFieldModel(
   z_hit_ = z_hit;
   z_rand_ = z_rand;
   sigma_hit_ = sigma_hit;
-  map_update_cspace(map, max_occ_dist);
+
+  // recompute cpace only when nescessary, i.e. if:
+  // - max_occ_dist changed
+  // OR
+  // - cspace was not computed yet, i.e. when map->max_occ_dist == 0.0
+  if (map->max_occ_dist != max_occ_dist || map->max_occ_dist == 0.0) {
```

Comment:
if we're going to rely on "== 0.0" - we should probably update map_alloc() (in map.c) to explicitly set max_occ_dist to 0.0 (it doesn't look like that currently happens).

Also, is 0.0 even in a valid max_occ_dist to be passed in here? It seems like not, in which case the first condition will cover us as long as we make sure an uninitialized map has 0.0 (which would be covered by updating map_alloc() in map.c)